File: number_series/algo_1.c
#include <stdlib.h>
#include <stdio.h>

void main()
{
    int *p_array, iterators;

    printf("Q1:  Write an algorithm to produce the first 15 numbers of this series: 1,1,2,3,5,8,13,21…\n");

    printf("Enter the number  of iterators: ");
    scanf("%d", &iterators);

    p_array = (int *)malloc(sizeof(int) * iterators); // allocate 50 ints

    for (int i = 0; i < iterators; i++)
    {
        if (i >= 2)
        {
            long current_value = p_array[i - 2] + p_array[i - 1];
            p_array[i] = current_value;
            printf("iteration number%d, sequenc value: %d\n", i, current_value);
        }
        else
        {
            p_array[i] = 1;
            printf("iteration number%d, sequenc value: %d\n", i, 1);
        }
        // p_array[i] = i + 1;
    }
}

File: number_series/Algo_1.py
''' Write an algorithm to produce the first 15 numbers of this series: 1,1,2,3,5,8,13,21…'''

''' 
*** mathematical equation > resp = n + (n+1) where n = index number

* Initial the index value with 0
* Set function defination
* add index with (index+1)
* interate this function for 15 times

*** Algorithm is n = index_value(n-1) + index_value(n-2) ***
'''


sequence_lis = []

for i in range(1, 15):
    if len(sequence_lis) >= 2:
        current_value = sequence_lis[-2:]
        current_value = current_value[0] + current_value[1]
        sequence_lis.append(current_value)
    else:
        sequence_lis.append(1)

sequence_lis = [str(x) for x in sequence_lis]
print(', '.join(sequence_lis))

